app/src/main/java/org/sopt/and/SignInActivity.kt

```diff
@@ -0,0 +1,318 @@
+    value: String,
+    onValueChange: (String) -> Unit,
+    showPassword: Boolean,
+    onVisibilityChange: () -> Unit
+) {
+    TextField(
+        value = value,
+        onValueChange = onValueChange,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(60.dp),
+        placeholder = { Text(text = "비밀번호", color = Color.Gray) },
+        colors = TextFieldDefaults.colors(
+            focusedContainerColor = Color.DarkGray,
+            unfocusedContainerColor = Color.DarkGray,
+            focusedTextColor = Color.White,
+            unfocusedTextColor = Color.White,
+            focusedIndicatorColor = Color.Transparent,
+            unfocusedIndicatorColor = Color.Transparent
+        ),
+        shape = RoundedCornerShape(5.dp),
+        visualTransformation = if (showPassword) VisualTransformation.None else PasswordVisualTransformation(),
+        trailingIcon = {
+            Text(
+                text = if (showPassword) "Hide" else "Show",
+                color = Color.White,
+                modifier = Modifier.clickable { onVisibilityChange() }
+            )
+        },
+        singleLine = true
+    )
+}
+
+@Composable
+fun SignInButton(text: String, onClick: () -> Unit) {
+    Button(
+        onClick = onClick,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(50.dp),
+        colors = ButtonDefaults.buttonColors(
+            containerColor = Color.Blue,
+            contentColor = Color.White
+        )
+    ) {
+        Text(text = text, fontSize = 15.sp)
+    }
+}
+
+@Composable
+fun ActionLinks(onSignUpClick: () -> Unit) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceEvenly
+    ) {
+        ActionText(text = "아이디 찾기", onClick = { })
+        Text(text = "|", color = Color.Gray)
+        ActionText(text = "비밀번호 재설정", onClick = { })
+        Text(text = "|", color = Color.Gray)
+        ActionText(text = "회원 가입", onClick = onSignUpClick)
+    }
+}
+
+@Composable
+fun ActionText(text: String, onClick: () -> Unit) {
+    Text(
+        text = text,
+        color = Color.Gray,
+        modifier = Modifier.clickable(onClick = onClick)
+    )
+}
+
+@Composable
+fun SNSButtons() {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.Center
+    ) {
+        val snsColors = listOf(Color.Yellow, Color.Blue, Color.Green, Color(0xFF113065), Color.White)
+        snsColors.forEach { color ->
+            Button(
+                onClick = {},
+                shape = CircleShape,
+                modifier = Modifier
+                    .padding(5.dp)
+                    .size(50.dp),
+                colors = ButtonDefaults.buttonColors(containerColor = color)
+            ) { }
+        }
+    }
+}
+
+fun showSnackbar(context: ComponentActivity, message: String) {
+    // Snackbar 메시지 표시를 위한 메서드
+    Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+}
```

Comment:
함수명과 주석은 SnackBar인데 구현된 코드는 Toast Message 같아요!! 확인 부탁드립니당~
그리고 함수화 깔끔하게 잘 하셨는데 아래처럼 context의 확장 함수로 만들 수도 있습니다!! 

```suggestion
fun Context.showToast(message: String, isShort: Boolean = true) {
    val duration = if (isShort) Toast.LENGTH_SHORT else Toast.LENGTH_LONG
    Toast.makeText(this, message, duration).show()
}
```

Reply:
스낵바 사용했다가 토스트로 수정하는 부분에서 놓쳤나봐요ㅠ 아직 context에 대해서 잘 몰라서 더 공부해오겠습니다!!!!